### DISCUSSION TITLE
Inject OpenSky credentials

### DISCUSSION THREAD

**Discussion by rmonstein** [xoolive/traffic#415] [Q&A] [answered]
I'm using Traffic in a container in the cloud and I would like to avoid to have my OpenSky credentials to access the historical database in a repository (i.e. have the config file in the repo).
Is there a way to inject/modify the credentials once the Traffic library is loaded?

**Comment by xoolive** [comment]
Would an environment variable make the job?

**Reply by rmonstein** [reply]
yes, that would have been my first choice. I can easily provide them to the container at startup. But what to do when I have the credentials in the container? How to I convince Traffic to use the new credentials as opposed to the ones in the config file?

**Reply by xoolive** [reply, marked as answer]
Would it work if you remove credentials from the config file and use the environment variables instead as described [here](https://open-aviation.github.io/pyopensky/credentials.html#fallback-configuration)?

**Reply by rmonstein** [reply]
Perfect, I should just read the documentation all the way to the end ;-)

**Reply by xoolive** [reply]
I shall not throw the first stone 😅
